locations/spiders/goodlifefitness.py

```diff
@@ -1,36 +1,37 @@
+import re
+
 import scrapy
+from scrapy.spiders import SitemapSpider
 
-from locations.items import GeojsonPointItem
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
 
 
-class GoodLifeFitnessSpider(scrapy.Spider):
+class GoodLifeFitnessSpider(SitemapSpider):
     name = "goodlifefitness"
     item_attributes = {"brand": "GoodLife Fitness", "brand_wikidata": "Q3110654"}
-    allowed_domains = ["www.goodlifefitness.com"]
-    start_urls = ("https://www.goodlifefitness.com/locations",)
-
-    def start_requests(self):
-        template = "https://www.goodlifefitness.com/api/club/filterclubsforcitypage"
-
-        headers = {
-            "Accept": "application/json",
-        }
-
-        yield scrapy.http.FormRequest(url=template, method="GET", headers=headers, callback=self.parse)
+    allowed_domains = ["goodlifefitness.com"]
+    sitemap_urls = ["https://www.goodlifefitness.com/sitemap.xml"]
+    sitemap_rules = [(r"/clubs/club.", "parse")]
 
     def parse(self, response):
-        store_data = response.json()
-        for store in store_data:
-            properties = {
-                "ref": store["ClubNo"],
-                "name": store["ClubName"],
-                "addr_full": store["Address1"],
-                "city": store["City"],
-                "state": store["Province"],
-                "postcode": store["PostalCode"],
-                "phone": store["Phone"],
-                "lat": float(store["Lat"]),
-                "lon": float(store["Long"]),
-            }
-
-            yield GeojsonPointItem(**properties)
+        template = "https://www.goodlifefitness.com/content/experience-fragments/goodlife/header/master/jcr:content/root/responsivegrid/header.GetClubsWithDetails.{}.false.true.20221127.json"
```

Comment:
I'm a little consigned that that's a date, and the spider will either break or return old data

Reply:
Since I’m looking for the id on the sitemap, I think it’s okay?

Reply:
Well the date is in the template url, hopefully closed locations would get dropped from the sitemap and then then the spider wouldn't scrape them. But would the API give data for new locations but with an old date from before they existed?

I think we're OK for now, we can revisit it when the spider breaks. However if I was writing this spider, I'd have just done it as a StructuredDataSpider (+ xpath the lat/lon), I know that's less metadata right now, but I think it's less likely to break, and I think they might add more data to that overtime, and then we'd get it automatically.

Reply:
Yes, I agree, I'm going to change it tonight